wps-cloudformation-template.yaml

```diff
@@ -753,6 +757,101 @@ Resources:
                   !Join ["", ["arn:aws:s3:::", !If [CreateEphemeralBucket, !Ref EphemeralS3Bucket, !FindInMap [EnvironmentMap, !Ref environment, BucketName]]]],
                   !Join ["", ["arn:aws:s3:::", !If [CreateEphemeralBucket, !Ref EphemeralS3Bucket, !FindInMap [EnvironmentMap, !Ref environment, BucketName]],"/*"]]
                 ]
+  ReloadLambdaCodeFunctionRole:
+    Type: 'AWS::IAM::Role'
+    Properties:
+      AssumeRolePolicyDocument:
+        Version: 2012-10-17
+        Statement:
+          - Effect: Allow
+            Principal:
+              Service:
+                - lambda.amazonaws.com
+            Action:
+              - 'sts:AssumeRole'
+      Path: /
+      ManagedPolicyArns:
+        - 'arn:aws:iam::aws:policy/service-role/AWSLambdaBasicExecutionRole'
+      Policies:
+        - PolicyName: UpdateVersion
+          PolicyDocument:
+            Version: 2012-10-17
+            Statement:
+              - Effect: Allow
+                Action:
+                  - 'lambda:UpdateFunctionCode'
```

Comment:
Can this access be narrowed at all to not be able to update any function?

The biggest concern I have about these custom resources is that they escape from the constraints of the template and are given access to arbitrary resources (as opposed to only the resources included in the template, addressed by reference).

Is there some way to limit this at least by the name of the function, or even better by reference (not sure if this creates a circular dependency however)?

Reply:
@lwgordonimos  Agree... I should have thought about it. 

Made the needed changes :+1:

Reply:
Nice! Makes me feel a lot less nervous about the custom resources.